pkg/controller/ingress/ingress_controller_test.go

```diff
@@ -62,18 +64,35 @@ func TestRouteMigration(t *testing.T) {
 	logf.SetLogger(logf.ZapLogger(true))
 
 	test := struct {
-		name        string
-		exRoute     *routev1.Route
-		wantName    string
-		removedName string
+		name            string
+		exRoute         *routev1.Route
+		noLabelRoute    *routev1.Route
+		otherLabelRoute *routev1.Route
```

Comment:
nit: Usually, these tests are written with a „state“ slice of entities and a „want“ slice of entities that we test against.

In this specific case we‘d pass all existing routes as „state“, call Reconcile, list all routes again and check what we‘ve got against the „want“ slice. Testify also has nice functions to assert that the resulting list has the same elements as the „want“ list.

That makes the test more generally useful and easier to extent with more edge cases.